### ISSUE TITLE
CLI: All cdktf-cli commands started failing today (workaround)

### ISSUE THREAD

**Issue by TreehouseFalcon** [hashicorp/terraform-cdk#3025]
### Expected Behavior

I have a `cdktf plan` execution in a Github Actions workflow and at some point today all executions started failing. I checked my local copy of the CLI and I am also completely unable to invoke it on a completely different operating system. I haven't touched any of our infrastructure code in at least a week, so I'm not sure what's going on. I've included the stack traces + my package.json in the gist section.

### Actual Behavior

I receive a file not found error for `cli-core/templates`. The stack traces when attempting to invoke any cdktf-cli commands are included in the Gist section.

### Steps to Reproduce

Windows:
1. Run `npm install --global cdktf-cli@latest`
2. Run `cdktf`

Github CI (Ubuntu):
1. Add a valid CDKTF configuration to a Github repo
2. Create a workflow that executes `cdktf plan` as one of the steps
3. Receive the error detailed below on on every dispatch

### Versions

language: typescript
cdktf-cli: 0.17.2
node: 18.16.0
cdktf: 0.17.2
constructs: 10.1.121
arch: x86
os: win 11/ubuntu latest

### Providers

N/A

### Gist

Windows: https://gist.github.com/TreehouseFalcon/c95fe0d293b953c8e4fdd1ff564ec2dd
Ubuntu/Github CI: https://gist.github.com/TreehouseFalcon/a0dc87e6b7cfdb3e9d576e1b02b2bd4f
My package.json: https://gist.github.com/TreehouseFalcon/ee34e0dea5e7366d3f075184d01602de

### Workarounds

Downgrade from v0.17.2 to v0.17.1.

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by TreehouseFalcon** [comment]
I forced our CI to use v0.17.1 instead of v0.17.2 and it seems to have completely fixed the issue. I see there was a release 5 hours ago so that must've been the culprit.

**Comment by dkwgit** [comment]
Ran into same error today.

**Comment by kination** [comment]
Depend on log, it is finding modules with node version `16.20`, but node version on description is `18.16`.
```
Error: ENOENT: no such file or directory, scandir '/opt/hostedtoolcache/node/16.20.1/x64/lib/node_modules/@cdktf/cli-core/templates'
    at Object.readdirSync (node:fs:1438:3)
```

which one is correct?

**Comment by dkwgit** [comment]
> Depend on log, it is finding modules with node version `16.20`, but node version on description is `18.16`.
> 
> ```
> Error: ENOENT: no such file or directory, scandir '/opt/hostedtoolcache/node/16.20.1/x64/lib/node_modules/@cdktf/cli-core/templates'
>     at Object.readdirSync (node:fs:1438:3)
> ```
> 
> which one is correct?

The failure happened for me in both node 20 and 16.

**Comment by JordanPavlic** [comment]
I ran into this same problem on windows 11 with node version 18.17: 

![image](https://github.com/hashicorp/terraform-cdk/assets/13093052/85d7e88d-3f62-4071-89a1-54f2e403dfd4)

I noticed that cdktf is expecting a file to present that isn't (@cdktf). See Below for my pictures:

![image](https://github.com/hashicorp/terraform-cdk/assets/13093052/bd0355cf-40d6-41ed-9811-53b186e611aa)


![image](https://github.com/hashicorp/terraform-cdk/assets/13093052/1366ebe1-8e83-4a63-af11-4c86e4ea6d29)

I fixed this after several hours of troubleshooting by downgrading from 0.17.2 to 0.17.1:
```
npm install --global cdktf-cli@0.17.1
```

**Comment by jsteinich** [comment]
Seems to have broken in https://github.com/hashicorp/terraform-cdk/commit/6babff72ba60a2231cfc751ebc842c1f04cc3c7d

**Comment by kination** [comment]
Also causing on MacOS.
After install, global package looks like:
```
~/.nvm/versions/node/v18.14.2/lib/node_modules (v0.39.3) » ls -la                                                                
total 0
drwxr-xr-x@  5 kination  staff  160 Jul 24 12:28 .
drwxr-xr-x@  4 kination  staff  128 Feb 21 13:05 ..
drwxr-xr-x@ 10 kination  staff  320 Jul 24 12:28 cdktf-cli
drwxr-xr-x@  8 kination  staff  256 Feb 21 13:05 corepack
drwxr-xr-x@ 12 kination  staff  384 Feb 21 13:05 npm

~/.nvm/versions/node/v18.14.2/lib/node_modules (v0.39.3) » cd cdktf-cli

~/.nvm/versions/node/v18.14.2/lib/node_modules/cdktf-cli (v0.39.3) » tree -I 'node_modules|cache|test_*'
.
├── README.md
├── ambient.d.ts
├── build.js
├── bundle
│   └── bin
│       ├── cdktf
│       ├── cdktf.js
│       ├── cdktf.js.map
│       └── cmds
│           ├── handlers.js
│           └── handlers.js.map
├── jest.config.js
├── package.json
└── src
    └── bin
        ├── cdktf
        └── cmds
            ├── helper
            │   ├── __tests__
            │   │   └── fixtures
            │   │       ├── foo.tfvars
            │   │       ├── hey-there.auto.tfvars
            │   │       └── terraform.tfvars
            │   └── render-ink.tsx
            └── ui
                ├── components
                │   ├── bottom-bars
                │   │   ├── approve.tsx
                │   │   ├── outputs.tsx
                │   │   ├── override.tsx
                │   │   └── status.tsx
                │   ├── outputs.tsx
                │   └── stream-view.tsx
                ├── deploy.tsx
                ├── destroy.tsx
                ├── diff.tsx
                ├── get.tsx
                ├── list.tsx
                ├── output.tsx
                ├── provider-list.tsx
                ├── synth.tsx
                └── watch.tsx
```

There's no `@cdktf/` or `template/`

**Comment by AndreyBespamyatnov** [comment]
We are having the same issue in the Azure DevOps pipelines (Ubuntu):

```
node:internal/fs/utils:350
    throw err;
    ^

Error: ENOENT: no such file or directory, scandir '/opt/hostedtoolcache/node/18.17.0/x64/lib/node_modules/@cdktf/cli-core/templates'
    at Object.readdirSync (node:fs:1527:3)
    at /opt/hostedtoolcache/node/18.17.0/x64/lib/node_modules/cdktf-cli/bundle/bin/cdktf.js:53:30496
    at /opt/hostedtoolcache/node/18.17.0/x64/lib/node_modules/cdktf-cli/bundle/bin/cdktf.js:1:226
    at /opt/hostedtoolcache/node/18.17.0/x64/lib/node_modules/cdktf-cli/bundle/bin/cdktf.js:53:30740
    at /opt/hostedtoolcache/node/18.17.0/x64/lib/node_modules/cdktf-cli/bundle/bin/cdktf.js:1:258
    at Object.<anonymous> (/opt/hostedtoolcache/node/18.17.0/x64/lib/node_modules/cdktf-cli/bundle/bin/cdktf.js:54:1932)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12) {
  errno: -2,
  syscall: 'scandir',
  code: 'ENOENT',
  path: '/opt/hostedtoolcache/node/18.17.0/x64/lib/node_modules/@cdktf/cli-core/templates'
}

```

**Comment by BrentSouza** [comment]
I was able to workaround it with the latest release by installing the package directly in the global namespace:

```sh
npm install --global @cdktf/cli-core
```

**Comment by kination** [comment]
As @BrentSouza has mentioned, seems some of package is not being installed with `npm install -g cdktf@0.17.2`.

**Comment by DanielMSchmidt** [comment]
We are working on a fix. The issue is we expect some files to be in a package next to the cdktf-cli but that is not there generally, the refactoring did work locally and in our integration tests, because the other package is there in these cases.

Please use the `npm install --global cdktf-cli@0.17.1` workaround until we release `0.17.3`

**Comment by ansgarm** [comment]
Hi everyone 👋 We just released [v0.17.3](https://github.com/hashicorp/terraform-cdk/releases/tag/v0.17.3) which fixes this bug. We've also improved how we run our end-to-end tests to ensure we catch this class of errors in CI.

**Comment by kination** [comment]
0.17.3 works well on MacOS
```
$ cdktf debug

language: null
cdktf-cli: 0.17.3
node: v18.14.2
terraform: 1.4.6
arch: arm64
os: darwin 22.2.0
```

Now package includes `templates` in `bundle/`
```
├── bundle
│   ├── bin
│   │   ├── cdktf
│   │   ├── cdktf.js
│   │   ├── cdktf.js.map
│   │   └── cmds
│   │       ├── handlers.js
│   │       └── handlers.js.map
│   └── templates
│       ├── csharp
│       │   ├── MainStack.cs
│       │   ├── MyTerraformStack.csproj
│       │   ├── Program.cs
│       │   ├── TestProgram.cs
│       │   ├── cdktf.json
│       │   ├── help
│       │   └── {{}}.gitignore
│       ├── go
│       │   ├── cdktf.json
│       │   ├── go.mod
│       │   ├── help
│       │   ├── main.go
│       │   ├── main_test.go
│       │   └── {{}}.gitignore
```
